### ISSUE TITLE
`xcode-project use-profiles` doesn't read inherited Info.plist setting

### ISSUE THREAD

**Issue by tomgilder** [codemagic-ci-cd/cli-tools#220]
Hello, when we added flavors to our Flutter project our iOS code signing broke. We are using automatic code signing to deploy to the App Store.

Flavors are Flutter's name for different build configurations, which create different build schemes in the Xcode project. We used [flutter_flavorizr](https://pub.dev/packages/flutter_flavorizr) to create the schemes, but our project then failed to build on Codemagic with this error:

```
Error (Xcode): "Runner" requires a provisioning profile. Select a provisioning profile in the Signing & Capabilities editor.
```

But everything in Xcode showed that the provisioning profile was correctly set for all build schemes. 

I tracked this down to the `xcode-project use-profiles` being unable to find the `INFOPLIST_FILE` build setting, with the command  outputting this in verbose mode:

```
Build configuration Release-flavor1 INFOPLIST_FILE is ''
Could not obtain bundle id value from base configuration reference
Failed to obtain bundle id for build_configuration 'Release-flavor1'
No bundle id found for target 'Runner'
```

This was due to the setting being inherited, rather than specified on each scheme.

Xcode showed the setting as being correct, because it was being inherited from the project:

<img width="1113" alt="KRClL" src="https://user-images.githubusercontent.com/756862/166926755-e5a66e6e-5d30-43a5-a18b-132119d1652f.png">

But we needed to change it to be specified for each scheme for `xcode-project use-profiles` to be able to read the value:

<img width="683" alt="qx8OO" src="https://user-images.githubusercontent.com/756862/166926903-0c2d253a-4d55-44fe-9c70-64bc0e94f3f6.png">

Ideally `xcode-project use-profiles` should be able to calculate the inherited value, or at least provide a good error, as this took me quite a long time to track down 😁 
 
We're not the only ones to face this, see [this Stack Overflow question](https://stackoverflow.com/questions/67386631/flutter-ios-signing-issue-runner-requires-a-provisioning-profile-with-the-pu/72127388).

**Comment by kristjanpirnpuu** [comment]
Had same issue with native iOS app. Inherited `PRODUCT_BUNDLE_IDENTIFIER` and some other settings from different xcconfig files for each build configuration.

Xcode 13 no longer creates Info.plist for new projects when using some of the project templates but generates it during build using values stored in project file. To get it working I had to stop using bundle id from xcconfig but hardcode it into settings in this case backed by xcodeproj.
[Xcode 13 release notes](https://developer.apple.com/documentation/xcode-release-notes/xcode-13-release-notes) Templates (68254857)

**Comment by priitlatt** [comment]
@tomgilder thanks for raising this issue and @kristjanpirnpuu thanks for extra details. We hope to work on it soon and we'll keep you posted regarding further progress.

**Comment by priitlatt** [comment]
@tomgilder and @kristjanpirnpuu, could you prepare a public repo containing a simple example project that suffers from this issue?

**Comment by priitlatt** [comment]
@tomgilder and @kristjanpirnpuu Alternatively, perhaps you could try out the possible fix I created in #232?

The patch can be installed as follows:
```bash
pip uninstall codemagic-cli-tools
pip install https://github.com/codemagic-ci-cd/cli-tools/archive/refs/heads/bugfix/use-profiles-infoplist-settings-inheritance.zip
```

**Comment by JonathanAJ** [comment]
Hello everyone. I'm having same's @tomgilder problems when I try build my project with flavors. I will try test @priitlatt fix. Thanks!

**Comment by JonathanAJ** [comment]
Hey guys, after fix made by @priitlatt my iOS workflow with flavors ran successfully!! 👍🏾 

But you must add the ` pip uninstall -y codemagic-cli-tools` with -y to accept uninstall.

Output after change:
```
--------------------------------------------------
Resolved bundle id '.......localdev' for build configuration 'Release-local_dev'
Using profile 'XC ..... localdev ios_app_store 000000000' (bundle id '.......localdev') for
	target 'Runner'
	build configuration 'Release-local_dev'
	bundle id '........localdev'
	specifier '00000-0000-0000-0000-0000000'
```

  Congratulations @priitlatt !! 👏🏾 👏🏾 👏🏾 👏🏾  When you can merge to oficial codemagic-cli?

**Comment by priitlatt** [comment]
@JonathanAJ thanks for checking it out and getting back with the results. I'll conduct some more testing on my part with other possible configurations to ensure that there is no regression with the introduced changes. All in all I hope to deploy it later today.

**Comment by priitlatt** [comment]
Fix is now released with [version 0.27.0](https://github.com/codemagic-ci-cd/cli-tools/releases/tag/v0.27.0). Sincere thanks to everyone who contributed!

**Comment by kristjanpirnpuu** [comment]
I was still using explicit build settings that was only way to build and after this update it started failing. Logs say no more than `xcodebuild` archive exit with status code 74. Then updated build settings to inherit from conf files but nothing changed.

Workspace itself contains 2 projects - app and framework both created with Xcode 13. App has implicit build dependency to framework by embedding it. Signing assets have been always successfully pulled from App Store Connect using credentials set from application environment variables(not shared in team). 

Before very first green build(prior [#232](https://github.com/codemagic-ci-cd/cli-tools/pull/232) merge) for `xcode-project use-profiles` had to specify `--custom-export-options` containing `distributionBundleIdentifier` that's bundle ID of app. Also had to set signing to manual and added signing identity, profile specifier directly into settings. Sounds like I'm doing something wrong but `CODE_SIGN_STYLE=MANUAL` is what I learned from [ios-automatic-code-signing-demo-project](https://github.com/codemagic-ci-cd/codemagic-sample-projects/blob/main/ios/ios-automatic-code-signing-demo-project/swiftly.xcodeproj/project.pbxproj). 

I'd love to have logic separated into framework and imported by different application targets. Also to have code signing automatic in regards of App Store Connect API as well as `CODE_SIGN_STYLE` because `CODE_SIGN_IDENTITY` and `PROVISIONING_PROFILE_SPECIFIER` can change automatically once certificate expires if I'm not mistaken. Current identity and profile contain some random integer generated by Apple. 

Last successful: https://codemagic.io/app/626a3b466248df417f54344e/build/62a72454ca62119431e7edcb
Next failed: https://codemagic.io/app/626a3b466248df417f54344e/build/62b9e93246756a3a4a8d8dd6
